### PR TITLE
extend gitignore

### PR DESCRIPTION
**Pull request by davydden**

### CODE REVIEW

**Comment by tjhei** [comment]
shouldn't these patterns start with ``/`` because they otherwise match files/folder in all subdirectories?

**Comment by davydden** [comment]
> shouldn't these patterns start with / because they otherwise match files/folder in all subdirectories?

done.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,14 @@
-build*
-code-gallery
+/*build*
+/code-gallery
+/doc/news/changes/changes.h.in
```
Comment (bangerth) [inline .gitignore]: why do you want to ignore this file? how does it get there?

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,14 @@
...
+
+# eclipse
+/.cproject
+/.project
+
+# vscode
```
Comment (bangerth) [inline .gitignore]: is this Visual Studio? Maybe use the more common name

File: .gitignore (hunk 1 above)
Comment (davydden) [reply .gitignore]: maybe a left-over from old days. Can remove

**Comment by davydden** [comment]
@bangerth @tjhei ping.

**Pull request merged**